app/Http/Controllers/FrontendTransportController.php

```diff
@@ -44,33 +44,44 @@ public function TrainAutocomplete(string $station): JsonResponse {
     public function TrainStationboard(Request $request): Renderable|RedirectResponse {
         $validated = $request->validate([
                                             'station'    => ['required'],
+                                            'station_id' => ['nullable', 'numeric'],
                                             'when'       => ['nullable', 'date'],
                                             'travelType' => ['nullable', new Enum(TravelType::class)]
                                         ]);
 
         $when = isset($validated['when']) ? Carbon::parse($validated['when']) : null;
 
         try {
-            $TrainStationboardResponse = TransportBackend::getDepartures(
-                stationQuery: $validated['station'],
+            //Per default: Use the given station query for lookup
+            $searchQuery = $validated['station'];
+
+            //If a station_id is given (=user is already on a stationboard) check if the user changed the query.
+            //If so: Use the given station string. Otherwise, use the station_id for lookup.
+            //This is to prevent that HAFAS fuzzy search return other stations (e.g. "Bern, Hauptbahnhof", Issue 1082)
+            if (isset($validated['station_id'])) {
+                $station = HafasController::getTrainStation($validated['station_id']);
+                if ($station->name === $validated['station']) {
+                    $searchQuery = $station->ibnr;
+                }
+            }
+            $stationboardResponse = TransportBackend::getDepartures(
+                stationQuery: $searchQuery,
                 when:         $when,
                 travelType:   TravelType::tryFrom($validated['travelType'] ?? null),
             );
+            return view('stationboard', [
+                                          'station'    => $stationboardResponse['station'],
+                                          'departures' => $stationboardResponse['departures'],
+                                          'times'      => $stationboardResponse['times'],
+                                          'latest'     => TransportController::getLatestArrivals(Auth::user())
+                                      ]
+            );
         } catch (HafasException $exception) {
             report($exception);
             return back()->with('error', __('messages.exception.generalHafas'));
         } catch (ModelNotFoundException) {
             return redirect()->back()->with('error', __('controller.transport.no-station-found'));
         }
-
-        return view('stationboard', [
-                                      'station'    => $TrainStationboardResponse['station'],
-                                      'departures' => $TrainStationboardResponse['departures'],
-                                      'times'      => $TrainStationboardResponse['times'],
-                                      'request'    => $request,
```

Comment:
Das haben Levin und ich vor 100 Jahren mal umgezogen, damit in der View nicht ständig `request()` aufgerufen wird. Unsicher, ob wir das jetzt nochmal umziehen wollen.

Reply:
@HerrLevin

Reply:
Weil @MrKrisKrisu sich nicht traut, zu antworten: Wir glauben, dass das wahrscheinlich inzwischen kein Problem mehr sein wird, weil die Funktion (wahrscheinlich) auch nur eine Variable returned. Meine Meinung ist "hm, mir egal. Willst du es lieber clean oder lieber performant?" und Kris will es lieber clean. 😅 

Das wird wahrscheinlich schon passen. Falls nicht, gibt es ein `git blame` und Kris darf es dann von Hand rückgängig machen. 😬 

---
Gemeinsam beraten und gesendet aus dem IC 396 in Ringsted St. auf Gleis 2.

Reply:
Okay, dann resolve ich das. Schönen Urlaub euch :)